syntax = "proto3";

enum SelectOption {
    GET = 0;
    GET_ALL = 1;
    ADD = 2;
    CREATE = 3;
    UPDATE = 4;
    REDUCE = 5;
    DELETE = 6;
}


message Product {
    int32 id =1;
    string product_id=2;
    string name = 3;
    string description =4;
    float price =5;
    bool is_available = 6;
    string message = 7;
    string error_message = 8;
    SelectOption option =9;
    int64 http_status_code = 10;
}

message ProductList {
    repeated Product products = 1;
}